src/main/scala/Square.scala

```diff
@@ -44,6 +44,11 @@ object Square:
     inline def file: File = File.of(s)
     inline def rank: Rank = Rank.of(s)
 
+    inline def xor(inline other: Square): Square =
```

Comment:
this is a beautiful transformation, but is this useful anywhere outside of epSquare calculation?

Reply:
probably not. better avoid adding this public api?

Reply:
probably just a function in `BinaryFen`, I feel like it's a bit unintuitive. But maybe it's just me.